keps/sig-auth/3257-cluster-trust-bundles/README.md

```diff
@@ -563,6 +578,52 @@ For example, if I maintain `example.com/my-signer`, I can use the following stra
   canary object first, and assess the health of the canary workloads.
 * Once I am satisfied that the change is safe, I edit the live object.
 
+### Publishing the kube-apiserver-serving Trust Bundle
+
+Today, the trust bundle that allows verifying kube-apiserver serving certificate(s)
+at its internal endpoints is distributed into every namespace using a configMap.
+This is so that it can be mounted along with the ServiceAccount token in order
+for the workloads to be able to communicate with the kube-apiserver.
+
+In the future, we should be able to replace mounting these configMaps in pods for
+for kube-apiserver trust with the projected volume from this feature, and so a
+ClusterTrustBundle API object will be created for all clusters:
+
+```yaml
+apiVersion: v1beta1
+kind: ClusterTrustBundle
+metadata:
+  generateName: kubernetes.io:kube-apiserver-serving:
+spec:
+  signerName: kubernetes.io/kube-apiserver-serving
+  trustBundle: "<... PEM CA ...>"
+```
+
+This object is managed by the Kubernetes Controller Manager in the existing
+`root-ca-certificate-publisher-controller`. It serves the same purpose
+and contains the same content as the `ca.crt` data in the `kube-root-ca.crt` configMap - to verify internal kube-apiserver
+endpoints. There is currently no in-tree signer designated for these purposes,
+and so the signer with name `kubernetes.io/kube-apiserver-serving` is introduced
+along with this bundle.
+
+This behavior is feature-gated by the `ClusterTrustBundle` KCM feature gate.
+
+#### Validation of the kube-apiserver-serving Trust Bundle
+
+Certificates in this bundle MUST NOT set the Extended Key Usage extension to "client auth".
+
+Certificates are valid for the bundle iff they fulfill at least one of the following
```

Comment:
these rules are for the certificates *issued* by signers in this bundle, not the CA certs in the bundle themselves

Reply:
> I'm mostly looking for the rules on the kube-apiserver certs (serving certs, with some guidance on expected SANs)

I see, that's the part I wasn't sure about. It's a bit odd situation with the signer because we won't be the ones doing the signing this time. Enforcing the rules gets a bit tricky this way, although a conformance test is a good idea on how to make sure our expectations on the signer are met.

I'll think about this section some more.

Reply:
I rewrote the whole section, it should now match the format of the signers from the k/website. I made the requirements on the signed certs' SANs a bit looser to allow for non-internal kube-apiserver hostnames, I wonder if that's something we may want to allow?